cli/tests/cfncluster-unittest.py

```diff
@@ -0,0 +1,294 @@
+    from io import StringIO
+
+from moto import mock_ec2, mock_cloudformation, mock_s3, mock_autoscaling
+
+import logging
+import re
+import boto3
+import unittest
+import configparser
+import os
+import stat
+import json
+
+test_log_stream = StringIO()
+config_file = 'cli/tests/config'
+with open('cloudformation/cfncluster.cfn.json') as f:
+    json_data = json.dumps(json.load(f))
+
+
+def config_logger_test():
+    logging.basicConfig(stream=test_log_stream, level=logging.INFO)
+
+
+def setup_configurations():
+    s3 = boto3.client('s3')
+    s3_conn = boto3.resource('s3')
+    s3.create_bucket(Bucket='us-east-1-cfncluster')
+    s3_conn.Object(
+        'us-east-1-cfncluster', 'cfncluster').put(Body=json_data)
+    template_url = s3.generate_presigned_url(
+        ClientMethod='get_object',
+        Params={
+            'Bucket': 'us-east-1-cfncluster',
+            'Key': 'cfncluster'
+        }
+    )
+
+    client = boto3.client('ec2', region_name='us-east-1')
+    instance = client.run_instances(ImageId='ami-1234abcd', MinCount=1, MaxCount=1)['Instances'][0]
+
+    vpc = client.create_vpc(CidrBlock='10.0.0.0/16', AmazonProvidedIpv6CidrBlock=True, DryRun=False)
+    subnet = client.create_subnet(AvailabilityZone=instance['Placement']['AvailabilityZone'],
+                                  CidrBlock='10.0.0.0/16',
+                                  Ipv6CidrBlock=vpc['Vpc']['Ipv6CidrBlockAssociationSet'][0]['Ipv6CidrBlock'],
+                                  VpcId=vpc['Vpc']['VpcId'])
+    config = configparser.ConfigParser()
+    config.read(config_file)
+    config.set('vpc public', 'vpc_id', subnet['Subnet']['VpcId'])
+    config.set('vpc public', 'master_subnet_id', subnet['Subnet']['SubnetId'])
+
+    open(config_file,'a').close()
+    os.chmod(config_file, stat.S_IRUSR | stat.S_IWUSR)
+    with open(config_file,'w') as cf:
+        config.write(cf)
+
+    return template_url
+
+
+class BaseArgs:
+    def __init__(self):
+        self.version = "0.0"
+        self.func = lambda x: x
+        self.config_file = "cli/tests/config"
+        self.region = "us-east-1"
+
+
+class CreateClusterArgs(BaseArgs):
+    def __init__(self, template_url, nowait):
+        BaseArgs.__init__(self)
+        self.cluster_name = "test_cluster"
+        self.template_url = template_url
+        self.norollback = True
+        self.nowait = nowait
+
+
+class UpdateClusterArgs(CreateClusterArgs):
+    def __init__(self, template_url, nowait, reset_desired):
+        CreateClusterArgs.__init__(self, template_url=template_url, nowait=nowait)
+        self.reset_desired = reset_desired
+
+
+class CFN_cluster_test(unittest.TestCase):
+    def setUp(self):
+        config_logger_test()
+
+    def test_cfn_cluster_version(self):
+        args = BaseArgs()
+        cfncluster.version(args)
+        log = test_log_stream.getvalue()
+        version = re.match(r"^INFO:\w+\.\w+:(\d+\.\d+\.\d+)$", log).group(1)
+        self.assertEqual(version, '1.5.1')
```

Comment:
Pull the current version from `setup.py` so we don't need to change multiple places.

Reply:
I have decided to take it from the json file instead of the python file.